crates/astria-celestia-client/src/blob_space.rs

```diff
@@ -0,0 +1,320 @@
+//! The blobs of data that are are submitted to celestia.
+
+use celestia_types::nmt::{
+    Namespace,
+    NS_ID_V0_SIZE,
+};
+use ed25519_consensus::{
+    Signature,
+    SigningKey,
+    VerificationKey,
+};
+use sequencer_types::ChainId;
+use sequencer_validation::InclusionProof;
+use serde::{
+    de::DeserializeOwned,
+    Deserialize,
+    Serialize,
+};
+use tendermint::{
+    block::{
+        Commit,
+        Header,
+    },
+    Hash,
+};
+
+/// Utility to create a v0 celestia namespace from the sha256 of `bytes`.
+#[must_use]
+pub fn celestia_namespace_v0_from_hashed_bytes(bytes: &[u8]) -> Namespace {
+    use sha2::{
+        Digest as _,
+        Sha256,
+    };
+    let mut hasher = Sha256::new();
+    hasher.update(bytes);
+    let result = hasher.finalize();
+    const _: () = assert!(NS_ID_V0_SIZE < 32);
+    let id = <[u8; NS_ID_V0_SIZE]>::try_from(&result[0..NS_ID_V0_SIZE])
+        .expect("must not fail as hash is always 32 bytes and NS_ID_V0_SIZE < 32");
+    Namespace::const_v0(id)
+}
+
+fn hash_json_serialized_bytes<T: Serialize>(val: &T) -> [u8; 32] {
+    use sha2::{
+        Digest,
```

Comment:
```suggestion
        Digest as _,
```

supernit for consistency with above